pkg/challenger/challenger_test.go

```diff
@@ -0,0 +1,104 @@
+// [✓] Setup a cluster
+// [✓] install crds on it
+// - run the server locally
+// - make requests to the server to see if we can get passphrases back
+package challenger_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("/challenge", func() {
+	Describe("Response", func() {
+		When("a matching sealedvolume exists", func() {
+			var tpmHash string
+			var sealedVolumeFile string
+
+			BeforeEach(func() {
+				tpmHash = fmt.Sprintf("%d", time.Now().UnixNano())
+				sealedVolumeFile = applySealedVolume(tpmHash)
+			})
+
+			AfterEach(func() {
+				deleteSealedVolume(sealedVolumeFile)
+				defer os.Remove(sealedVolumeFile)
+			})
+
+			It("returns the passphrase", func() {
+				Expect(true).To(BeTrue())
+			})
+		})
+
+		When("a matching sealedvolume doesn't exist", func() {
+			It("returns an error", func() {
+				Expect(true).To(BeTrue())
```

Comment:
At this point, we need to run the challenger client and make a request to the server for a passphrase. The request is constructed using an actual TPM and I don't see an easy way around that. This means, I need to run the command inside a VM with an emulated TPM (:face_exhaling:). This PR is growing too fast.

Reply:
I switched to unit tests for now. They are faster simpler and they do the job. We will introduce an end-to-end suite in another PR